ios/brave-ios/Sources/BraveWallet/Extensions/AssetRatioServiceExtensions.swift

```diff
@@ -7,140 +7,59 @@ import BraveCore
 import Foundation
 
 extension BraveWalletAssetRatioService {
-  typealias PricesResult = (assetPrices: [BraveWallet.AssetPrice], failureCount: Int)
-
-  /// Filter out empty strings, make unique, and ensure lowercase
-  private func uniqueNonEmptyPriceIds(_ priceIds: [String]) -> [String] {
-    Array(Set(priceIds.filter { !$0.isEmpty }.map { $0.lowercased() }))
-  }
-
-  /// Fetch the price for a list of assets to a list of assets over a given timeframe.
-  /// If the main call for all asset prices fails, will fetch the price for each asset individually, so one failure will not result in a failure to fetch all assets.
-  @MainActor func priceWithIndividualRetry(
-    _ fromAssets: [String],
-    toAssets: [String],
-    timeframe: BraveWallet.AssetPriceTimeframe
-  ) async -> PricesResult {
-    let uniquePriceIds = uniqueNonEmptyPriceIds(fromAssets)
+  /// Fetches the prices for a given list of `AssetPriceRequest`, giving a list of `AssetPrice`.
+  /// Returns an empty array in case no prices are found.
+  @MainActor func fetchPrices(
+    for requests: [BraveWallet.AssetPriceRequest],
```

Comment:
i think it would be easier to make function take a list of tokens and put the map that transform token to BraveWallet.AssetPriceRequest here so that you don't need to map everywhere when you call this function. because most of the time we have a list of token in hand.